### PR TITLE
Add esp-aves2-sed-birdcode BEATs encoder as an official model

### PR DESCRIPTION
**Pull request by GaganNarula**
Registers the encoder-only BEATs backbone from the `sed-birdcode-ablation-ssl_beats_clip_pseudo` run, so it shows up in `list_models()` and loads by name:

```python
from avex import load_model
backbone = load_model("esp_aves2_sed_birdcode_beats_encoder", device="cuda")
```

`list_models()` goes from 10 to 11 entries.

## What was published

The source checkpoint stored its backbone under an `encoder._model.backbone.*` prefix and carried a frame-level SED head, `Linear(6144, 7475)`. Publishing involved two changes, both lossless for the backbone:

1. Key prefix `encoder._model.backbone.*` → `backbone.*`, matching `avex.models.beats_model.Model`.
2. The head was dropped — 252 backbone tensors kept, 2 head tensors removed.

`load_model` reports **252/252 params matched, 0 unexpected** and auto-selects embedding mode (no classifier in the checkpoint → `num_classes=None`). Backbone weights are **bitwise identical** to the source, verified by a `strict=True` load and by comparing forward-pass outputs.

Artifacts live in GCS at `gs://representation-learning/esp-avex/models/esp-aves2-sed-birdcode-ablation-ssl-beats-clip-pseudo-encoder/` and on the Hub at [`EarthSpeciesProject/esp-aves2-sed-birdcode-ablation-ssl-beats-clip-pseudo-encoder`](https://huggingface.co/EarthSpeciesProject/esp-aves2-sed-birdcode-ablation-ssl-beats-clip-pseudo-encoder).

## Input contract — not the AVEX defaults

| Setting | Value |
|---|---|
| `audio_config.sample_rate` | **32000** |
| `audio_config.target_length_seconds` | **5.0** |
| `init_config.sample_frequency` | **16000** — do not change |

`sample_frequency` is a parameter of the fbank extractor, not a declaration of the input rate. It fixes the window (400 samples), hop (160), and mel filterbank (257 × 128), all frozen in the checkpoint as `backbone.fbank.window` `(400,)` and `backbone.fbank.mel_fb` `(257, 128)`. Setting it to 32000 rebuilds them at `(800,)` and `(513, 128)` and **the checkpoint fails to load**.

So 32 kHz audio runs through a 16 kHz fbank: windows span 12.5 ms of real time and content at true frequency *f* is analysed at *f*/2. 5.0 s yields 496 tokens (62 time × 8 freq) of dim 768. Two failure modes: resampling input to 16 kHz loads but silently degrades; changing `sample_frequency` won't load at all. Documented in the YAML and the model card.

## Registry key naming

The key is deliberately shorter than the Hub repo / GCS folder name, which would otherwise be a 61-character identifier for callers to type — and which pushed the checksum/fingerprint table lines past the 120-column ruff limit. `checkpoint_path` in the YAML carries the full repo name.

## Blockers before merge

- [ ] **Two fingerprint profiles are placeholders.** `test_reference_table_covers_all_official_hf_models` requires an entry in every profile, and `torch_2_5_0` / `torch_2_6_0` cannot be computed from a torch 2.11 environment. Both currently hold the torch 2.11 value and are marked `TODO`; they will **not** match. Regenerate from those environments:
  ```
  uv run python scripts/regenerate_official_model_output_fingerprints.py --profile torch_2_5_0
  uv run python scripts/regenerate_official_model_output_fingerprints.py --profile torch_2_6_0
  ```
  `torch_2_11_0` was computed and verified locally.
- [ ] **The Hub repo is private.** Network-dependent tests (`test_official_models_checksums`, output regression) will skip without a token. Make the repo public, or accept the skips.
- [ ] **Model card has placeholders.** License is `TBA` (front matter carries `cc-by-nc-sa-4.0` copied from the sibling purely to satisfy Hub validation — it is **not** a confirmed license). Training data, hyperparameters, evaluation, compute, and citation are all `TBA`. This is an ablation checkpoint with no benchmark numbers; the card says so explicitly.

## Verification performed

- `strict=True` load of the source backbone: 252/252, zero missing, zero unexpected
- `load_model` via local path, `gs://` (`.pt` and `.safetensors`), `hf://`, and registry name — all 252/252 and bitwise identical to the source
- `convert_to_safetensors.py --verify full`: 252 tensors match
- `assert_safetensors_has_weights`: passes
- Offline table validation: all three fingerprint profiles and the checksum table match the official HF model set exactly

The full pytest suite was **not** run locally.

### CODE REVIEW

**Review by nkundiushuti — APPROVED** [review]
It seems ok

**Comment by GaganNarula** [comment]
> It seems ok

hi marius, thanks for reviewing. A couple of questions before i merge this:

1. Should this model be in the official docs ? i believe you have some method of deciding how a model should be chosen for a project but i have no idea what this one would be used for.
2. what license should i add to the hf repo model card ? cc by nc sa 4.0 ?

**Comment by nkundiushuti** [comment]
> > It seems ok
> 
> hi marius, thanks for reviewing. A couple of questions before i merge this:
> 
> 1. Should this model be in the official docs ? i believe you have some method of deciding how a model should be chosen for a project but i have no idea what this one would be used for.
> 2. what license should i add to the hf repo model card ? cc by nc sa 4.0 ?

Thanks for the heads up and the work. The license should be according to the training data, I think the one you suggested is correct. 

No 1 I think you are referring to the model selector. I would need to redesign it and couple it with a more global leaderboard.
